tools/azure-devtools/src/azure_devtools/perfstress_tests/random_stream.py

```diff
@@ -29,7 +29,34 @@ def read(self, size=None):
             self._base_data = get_random_bytes(e)
             self._base_data_length = e
         self._remaining = self._remaining - e
+        self._position += e
         return self._base_data[:e]
+    
+    def tell(self):
+        return self._position
+
+    def seek(self, index):
+        self._position = index
 
     def remaining(self):
-        return self._remaining
\ No newline at end of file
+        return self._remaining
+
+
+class WriteStream:
```

Comment:
This is probably fast enough, but I wonder if it would still work and be even faster if it "cheated" and didn't bother to track length and position?  Where `write()` is just a no-op, `seekable` is false, `tell` always returns 0, etc.

Reply:
With this implementation, I also think all tests could re-use a single static WriteStream.

Reply:
But if the `WriteStream` needs to support length and seeking for correctness, then each test instance will need its own, but maybe it could just "reset" the stream between operations instead of creating a new one.

Reply:
Yeah unfortunately the SDK makes use of seekable and tell for multi-threaded downloads - so I think to get the best indication of download perf we should include testing of those mechanisms.
Additionally I could add a "reset" feature to the stream to both this and the RandomStream - but I think I will use a dedicated `reset` function that will be run outside of the clock for each test iteration.